README.md

```diff
@@ -48,3 +48,151 @@ To test with Nextflow for development purpose:
+
+#### Usage
+
+When using this function in your pipeline, you can provide the name of a JSON Schema file. It defaults to '`nextflow_schema.json`'.
+
+```
+validateParameters('custom_nextflow_schema.json')
+```
+
+#### Options
+
+There are two specific params that affect the behavior of this function:
+
+- fail_unrecognised_params
+
+When parameters which are not specified in the JSON Schema are provided, the parameter validation returns a `WARNING`. To force the pipeline execution fail returning an `ERROR` instead, you can provide the  `fail_unrecognised_params` parameter.
+```
+nextflow run my_pipeline --fail_unrecognised_params
+```
+or specifying it in the configuration file
+```
+params.fail_unrecognised_params = true
+```
+
+- lenient_mode
+
+The lenient mode of JSON Schema validation tries to cast parameters to their correct type. For example, providing an integer as a string won't fail when using such mode. You can find more information [here](https://github.com/everit-org/json-schema#lenient-mode).
+
+```
+nextflow run my_pipeline --lenient_mode
+```
+or specifying it in the configuration file
+```
+params.lenient_mode = true
+```
+
+### paramsHelp
+
+This function prints a help message with the command to run a pipeline and the available parameters.
+
+> **_NOTE:_**  paramsHelp() doesn't stop pipeline execution after running. You must add this into your pipeline code if it's the desired functionality.
+
+#### Usage
+
+This function required an argument providing the typical command used to run the pipeline. It can also accept the name of a JSON Schema file [default = '`nextflow_schema.json`'].
+
+In this example we are executing the function if the parameter `help` is provided and ending the execution afterwards.
+```
+if (params.help) {
+   def String command = "nextflow run my_pipeline --input input_file.csv --output output_directory"
+   log.info paramsHelp(command, 'custom_nextflow_schema.json')
+   System.exit(0)
+}
+```
+
+#### Options
+
+- show_hidden_params
+
+Params that are defined to be hidden in the JSON Schema are not shown in the help message. In order to show these parameters you can use the `show_hidden_params` parameter.
+
+```
+nextflow run my_pipeline --help --show_hidden_params
+```
+or specifying it in the configuration file
+```
+params.show_hidden_params = true
+```
+
+- Show the complete information for one parameter
+
+By default, when printing the help message only a selection of attributes are printed: type of the variable, accepted options "enums", description and default value. In order to print the complete information for a single parameter, you can pass the parameter name to `--help`:
+
+```
+nextflow run my_pipeline --help param_name
+```
+
+### paramsSummaryMap
+
+This function returns a Groovy Map summarizing parameters/workflow options used by the pipeline.
+
+> **_NOTE:_**  paramsSummaryMap() will return only the provided parameters that differ from the default values.
```

Comment:
Get [free fancy GitHub formatting](https://github.com/community/community/discussions/16925) with this syntax:
```suggestion
> **Note**
> `paramsSummaryMap()` will return only the provided parameters that differ from the default values.
```